File: dataset/human/c/file_139.c
#include<stdio.h>

int main(){
int r,g,b,n,cnt=0;
scanf("%d %d %d %d",&r,&g,&b,&n);
for(int i=n/r;i>=0;i--){
for(int j=(n-i*r)/g;j>=0;j--){
if((n-i*r-j*g)%b==0)cnt++;
}
}
printf("%d",cnt);
return 0;
}

File: dataset/human/python/file_139.py
import sys
input = sys.stdin.readline

r, g, b, n = map(int, input().split())



cnt = 0
for i in range(int(n/r) + 1):
    for j in range(int((n-r*i)/g) + 1):

        if (n - (r*i + g*j))%b==0 and n - (r*i + g*j) >= 0:
            cnt += 1
print(cnt)
